### PR TITLE
[openui5] Update the definition files for OpenUI5 1.112

### PR DESCRIPTION
**Pull request by akudev**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).  → as detailed in the [previous pull request for 1.90](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/53680), some are unavoidable at this point
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: → no specific URL about the changes, it's just a new release: https://sdk.openui5.org/1.112.0
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

@ DefinitelyTyped maintainers: the lint rule "no-mergeable-namespace" had to be disabled because of one violation which had to be newly introduced: the respective d.ts file is composed from pieces from different sources where both have to amend the same module. (Note: the entire type definitions are generated from JSDoc, so seemingly "easy" manual tweaks cannot be done easily.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  **types/openui5/sap.ui.integration.d.ts#L1836** - The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  **types/openui5/sap.ui.table.d.ts#L535** - The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.
  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 03d801992efe998112a4417d95c160a18b01278b
</p>

File: types/openui5/sap.ui.webc.main.d.ts [hunk 1]
```diff
@@ -37145,7 +37142,7 @@ declare module "sap/ui/webc/main/TimePicker" {
       /**
        * A Java Script date object to be formatted as string
        */
-      date: Date
+      date: object
```
Comment (codeworrior) [inline types/openui5/sap.ui.webc.main.d.ts:37145]: this looks like a regression?

Comment (codeworrior) [reply types/openui5/sap.ui.webc.main.d.ts:37145]: ...in the original source, I mean

Comment (codeworrior) [reply types/openui5/sap.ui.webc.main.d.ts:37145]: multiple times in the same lib

**Comment by akudev** [comment]
> Fails
> 🚫	
> **types/openui5/sap.ui.integration.d.ts#L1836** - The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.
> 
> 🚫	
> **types/openui5/sap.ui.table.d.ts#L535** - The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.
> 
> Generated by 🚫 [dangerJS](https://danger.systems/js) against [03d8019](https://github.com/DefinitelyTyped/DefinitelyTyped/commit/03d801992efe998112a4417d95c160a18b01278b)

That's tabs inside code sample snippets inside comments... but as it only seems to have happened twice, we'll try to change those. To scan in place to prevent future occurrences, though.

File: types/openui5/sap.ui.webc.main.d.ts (hunk 1 above)
Comment (akudev) [reply types/openui5/sap.ui.webc.main.d.ts:37145]: Yes, but locally in the source file, not in the generation:

In https://github.com/SAP/openui5/commit/2a5a857e56cdfbe51261c5d6dcf696ae0cd9f40f the return type was manually improved inside the openui5 repo,
but in https://github.com/SAP/openui5/commit/502191ae5bd04a31341a649852285fbc0e9da6c5 the new Web Components version was copied over, reverting the return type to the previous "object".

We need to teach developers to fix those types in the origin repo, not in openui5.

When changing the whitespace, I will also fix this and possibly 1-2 other occurrences.

**Review by codeworrior — APPROVED** [review]
LGTM

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
the config change (rule exclusion) is OK, as this is not feasible to use exclusion inline for those generated files.

**Comment by akudev** [comment]
Ready to merge

**Pull request merged**